From 918feacacb650cae09d86b42998dc75185eed114 Mon Sep 17 00:00:00 2001
From: Marc Olberding <molberding@nvidia.com>
Date: Thu, 20 Nov 2025 15:26:14 -0800
Subject: [PATCH 8/8] entity-manager: fix index bug in perform_scan.cpp

Previously, matched names were not being replaced
before they hit dbus. This caused names that depend on index
to only allow one instance of a given device, as $index was being
formatted to __index rather than the subsistuted value.

Tested: on nvl32-obmc startup entity-manager
before
```
busctl tree xyz.openbmc_project.EntityManager
`- /xyz
  `- /xyz/openbmc_project
    |- /xyz/openbmc_project/EntityManager
    `- /xyz/openbmc_project/inventory
      `- /xyz/openbmc_project/inventory/system
        `- /xyz/openbmc_project/inventory/system/board
          |- /xyz/openbmc_project/inventory/system/board/Nvidia_Alon8_HPM
          | `- /xyz/openbmc_project/inventory/system/board/Nvidia_Alon8_HPM/Nvidia_Alon8_HPM_FRU
          |- /xyz/openbmc_project/inventory/system/board/Nvidia_IMGX_ConnectX8_SuperNIC_Switch
          | `- /xyz/openbmc_project/inventory/system/board/Nvidia_IMGX_ConnectX8_SuperNIC_Switch/Nvidia_IMGX_ConnectX8_SuperNIC_Switch_FRU
          `- /xyz/openbmc_project/inventory/system/board/Nvidia_RTX_PRO_6000_Blackwell__index
            `- /xyz/openbmc_project/inventory/system/board/Nvidia_RTX_PRO_6000_Blackwell__index/Nvidia_RTX_PRO_6000_FRU
```

After:
```
busctl tree xyz.openbmc_project.EntityManager
`- /xyz
  `- /xyz/openbmc_project
    |- /xyz/openbmc_project/EntityManager
    `- /xyz/openbmc_project/inventory
      `- /xyz/openbmc_project/inventory/system
        `- /xyz/openbmc_project/inventory/system/board
          |- /xyz/openbmc_project/inventory/system/board/Nvidia_Alon8_HPM
          | `- /xyz/openbmc_project/inventory/system/board/Nvidia_Alon8_HPM/Nvidia_Alon8_HPM_FRU
          |- /xyz/openbmc_project/inventory/system/board/Nvidia_IMGX_ConnectX8_SuperNIC_Switch
          | `- /xyz/openbmc_project/inventory/system/board/Nvidia_IMGX_ConnectX8_SuperNIC_Switch/Nvidia_IMGX_ConnectX8_SuperNIC_Switch_FRU
          |- /xyz/openbmc_project/inventory/system/board/Nvidia_RTX_PRO_6000_Blackwell_1
          | `- /xyz/openbmc_project/inventory/system/board/Nvidia_RTX_PRO_6000_Blackwell_1/Nvidia_RTX_PRO_6000_FRU
          |- /xyz/openbmc_project/inventory/system/board/Nvidia_RTX_PRO_6000_Blackwell_2
          | `- /xyz/openbmc_project/inventory/system/board/Nvidia_RTX_PRO_6000_Blackwell_2/Nvidia_RTX_PRO_6000_FRU
          |- /xyz/openbmc_project/inventory/system/board/Nvidia_RTX_PRO_6000_Blackwell_3
          | `- /xyz/openbmc_project/inventory/system/board/Nvidia_RTX_PRO_6000_Blackwell_3/Nvidia_RTX_PRO_6000_FRU
          |- /xyz/openbmc_project/inventory/system/board/Nvidia_RTX_PRO_6000_Blackwell_4
          | `- /xyz/openbmc_project/inventory/system/board/Nvidia_RTX_PRO_6000_Blackwell_4/Nvidia_RTX_PRO_6000_FRU
          |- /xyz/openbmc_project/inventory/system/board/Nvidia_RTX_PRO_6000_Blackwell_5
          | `- /xyz/openbmc_project/inventory/system/board/Nvidia_RTX_PRO_6000_Blackwell_5/Nvidia_RTX_PRO_6000_FRU
          |- /xyz/openbmc_project/inventory/system/board/Nvidia_RTX_PRO_6000_Blackwell_6
          | `- /xyz/openbmc_project/inventory/system/board/Nvidia_RTX_PRO_6000_Blackwell_6/Nvidia_RTX_PRO_6000_FRU
          |- /xyz/openbmc_project/inventory/system/board/Nvidia_RTX_PRO_6000_Blackwell_7
          | |- /xyz/openbmc_project/inventory/system/board/Nvidia_RTX_PRO_6000_Blackwell_7/GPU_6
          | `- /xyz/openbmc_project/inventory/system/board/Nvidia_RTX_PRO_6000_Blackwell_7/Nvidia_RTX_PRO_6000_FRU
          `- /xyz/openbmc_project/inventory/system/board/Nvidia_RTX_PRO_6000_Blackwell_8
            |- /xyz/openbmc_project/inventory/system/board/Nvidia_RTX_PRO_6000_Blackwell_8/GPU_7
            |- /xyz/openbmc_project/inventory/system/board/Nvidia_RTX_PRO_6000_Blackwell_8/GPU_7SMA
            `- /xyz/openbmc_project/inventory/system/board/Nvidia_RTX_PRO_6000_Blackwell_8/Nvidia_RTX_PRO_6000_FRU
```

Change-Id: Ide8e6c1bee7870216cdda6a00718b4c82aa23671
Signed-off-by: Marc Olberding <molberding@nvidia.com>
---
 src/entity_manager/perform_scan.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/entity_manager/perform_scan.cpp b/src/entity_manager/perform_scan.cpp
index c126c0f..c54b42e 100644
--- a/src/entity_manager/perform_scan.cpp
+++ b/src/entity_manager/perform_scan.cpp
@@ -432,7 +432,7 @@ static void applyTemplateAndExposeActions(
 {
     nlohmann::json::object_t* exposeObj =
         value.get_ptr<nlohmann::json::object_t*>();
-    if (exposeObj != nullptr)
+    if (exposeObj == nullptr)
     {
         return;
     }
@@ -536,6 +536,8 @@ void scan::PerformScan::updateSystemConfiguration(
         std::string deviceName = generateDeviceName(
             usedNames, dbusObject, foundDeviceIdx, *name, replaceStr);
 
+        record["Name"] = deviceName;
+
         usedNames.insert(deviceName);
 
         for (auto& keyPair : record)
-- 
2.47.3

